tests/kernel/spinlock/src/main.c

```diff
@@ -82,6 +83,7 @@ static void bounce_once(int id, bool trylock)
 			break;
 		}
 
+		k_busy_wait(10);
```

Comment:
We should add comments about these new delays - as it's completely not clear what we are trying to wait. We are adding these two delays to the part of code where spinlocked is locked - so by the first look it should only make the situation worse.

Reply:
PS: I've looked into `test_trylock` - not sure that it's correct. We are expecting that the spinlock would be always locked sucessfuly with the `test_trylock` but we are relying on micro-scheduling with busy wait. IMO that won't work reliable, especially on simulatiors.

Reply:
Does it break some of your tests? The test is about 'fairness', not about getting lock.